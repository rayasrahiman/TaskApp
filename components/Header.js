import React from 'react';
import {
  View,
  Text,
  Image,
  StyleSheet,
  Dimensions,
  TouchableOpacity,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import {Colors} from '../constants/colors';

export default function Header({
  dashboard,
  headerRight,
  onPress,
  title,
  ...headerProps
}) {
  return (
    <View
      {...headerProps}
      style={{
        ...styles.container,
        ...(dashboard && {
          marginTop: Dimensions.get('window').height < 800 ? -hp(11) : -hp(5),
        }),
      }}>
      <View style={styles.imgCont}>
        <Image
          source={require('../assets/Logo.png')}
          style={styles.img}
          resizeMode="contain"
        />
      </View>
      {headerRight && (
        <TouchableOpacity onPress={onPress} style={styles.btnCont}>
          <Text style={styles.btnTxt}>{title}</Text>
        </TouchableOpacity>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    // justifyContent: 'center',
    alignItems: 'center',
    borderBottomColor: Colors.gray,
    borderBottomWidth: wp(0.3),
    width: Dimensions.get('window').width,
    height: hp(15),
  },
  imgCont: {
    marginLeft: wp(40),
  },
  img: {
    width: wp(15),
    height: hp(15),
  },
  btnCont: {
    marginRight: wp(5),
  },
  btnTxt: {
    fontSize: hp(3),
    color: Colors.lightBlue,
  },
});
