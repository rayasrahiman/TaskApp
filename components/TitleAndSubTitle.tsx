import {View, Text, StyleSheet, Platform} from 'react-native';
import React from 'react';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';

export type Props = {
  title: string;
  subTitle: string;
  titleContProp: any;
  titleTxtProp: any;
  subTitleContProp: any;
  emailSubTitleContProp: any;
  subTitleTxtProp: any;
  viewPager: boolean;
  email: boolean;
  emailVal: string;
};

const TitleAndSubTitle: React.FC<Props> = ({
  title,
  subTitle,
  titleContProp,
  titleTxtProp,
  subTitleContProp,
  emailSubTitleContProp,
  subTitleTxtProp,
  viewPager,
  email,
  emailVal,
  ...titleAndSubTitleProp
}) => {
  return (
    <View {...titleAndSubTitleProp}>
      <View
        style={{
          ...{
            marginHorizontal: viewPager
              ? 0
              : Platform.OS === 'android'
              ? wp(5)
              : wp(3),
          },
          ...titleContProp,
        }}>
        <Text style={{...styles.titleTxt, ...titleTxtProp}}>{title}</Text>
      </View>
      <View
        style={{
          ...{
            marginHorizontal: viewPager
              ? 0
              : Platform.OS === 'android'
              ? wp(5)
              : wp(3),
          },
          ...subTitleContProp,
        }}>
        <Text
          style={{
            ...styles.subTitleTxt,
            ...{color: Colors.grayishBlack},
            ...subTitleTxtProp,
          }}>
          {subTitle}
        </Text>
      </View>
      {email && (
        <View
          style={{
            ...{
              marginHorizontal: viewPager
                ? 0
                : Platform.OS === 'android'
                ? wp(5)
                : wp(3),
              marginBottom: hp(2),
            },
            ...emailSubTitleContProp,
          }}>
          <Text
            style={{
              ...styles.subTitleTxt,
              ...{color: Colors.black},
              ...subTitleTxtProp,
            }}>
            {emailVal}
          </Text>
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  titleTxt: {
    color: Colors.darkBlue,
    fontWeight: 'bold',
    fontSize: Platform.OS === 'android' ? hp(5) : hp(4),
  },
  subTitleTxt: {
    color: Colors.grayishBlack,
    fontWeight: 'bold',
    fontSize: Platform.OS === 'android' ? hp(3) : hp(2),
  },
});

export default TitleAndSubTitle;
