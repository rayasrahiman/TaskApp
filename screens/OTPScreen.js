import React, {useState} from 'react';
import {
  View,
  Text,
  StyleSheet,
  Dimensions,
  Platform,
  KeyboardAvoidingView,
  ScrollView,
  Pressable,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import AsyncStorage from '@react-native-async-storage/async-storage';

import Header from '../components/Header';
import Input from '../components/Input';
import TitleAndSubTitle from '../components/TitleAndSubTitle';
import Button from '../components/Button';
import I18n from '../languages/i18n';
import OTP from '../components/OTP';
import {Colors} from '../constants/colors';
import ModalComp from '../components/ModalComp';

export default function OTPScreen({navigation, route}) {
  const [disable, setDisable] = useState(true);
  const [error, setError] = useState({
    newInputError: false,
    confirmInputError: false,
    OTPCodeError: false,
  });
  const [confirmNew, setConfirmNew] = useState(false);
  const [OTPCode, setOTPCode] = useState('');
  const [newInput, setNewInput] = useState('');
  const [confirmInput, setConfirmInput] = useState('');
  const [modalVisible, setModalVisible] = useState(false);

  const regex =
    /^(\S)(?=.*[0-9])(?=.*[A-Z])(?=.*[a-z])(?=.*[~`!@#$%^&*()--+={}\[\]|\\:;"'<>,.?/_₹])[a-zA-Z0-9~`!@#$%^&*()--+={}\[\]|\\:;"'<>,.?/_₹]{8,20}$/;

  const newInputvalidate = text => {
    setNewInput(text);
    setDisable(!regex.test(text));
  };
  const confirmInputvalidate = text => {
    setConfirmInput(text);
    // setDisable(!regex.test(text));
  };

  const onChangeOTP = value => {
    setOTPCode(value);
  };

  const redirect = async () => {
    const array = await AsyncStorage.getItem('users');
    const arr = array ? JSON.parse(array) : [];

    if (!disable && confirmInput === newInput && OTPCode === '123456') {
      const user = arr.filter(item =>
        item.email === route.params.email ? (item.password = newInput) : item,
      );
      AsyncStorage.setItem('users', JSON.stringify(user));
      setConfirmNew(true);
      setModalVisible(true);
    } else {
      setConfirmNew(false);
      setError({
        newInputError: !disable ? false : true,
        confirmInputError: confirmInput === newInput ? false : true,
        OTPCodeError: OTPCode === '123456' ? false : true,
      });
    }
  };

  const modalbtnOnPress = () => {
    setModalVisible(!modalVisible);
    navigation.navigate('SignIn', {
      email: route.params.email,
    });
  };
  return (
    <View style={styles.mainWrapper}>
      <Header />
      <ScrollView>
        <ModalComp
          onRequestClose={() => {
            setModalVisible(!modalVisible);
          }}
          iconName="checkmark-circle"
          title={I18n.t('PassChangedTitle')}
          subTitle={I18n.t('PassChangedSubTitle')}
          onPress={modalbtnOnPress}
          visible={modalVisible}
          btnTxt={I18n.t('Continue')}
        />
        <KeyboardAvoidingView behavior="height" style={styles.container}>
          <ScrollView>
            <TitleAndSubTitle
              title={I18n.t('OTPTitle')}
              subTitle={I18n.t('OTPSubtitle')}
              subTitleContProp={styles.subTitleCont}
            />
            <OTP
              onCodeChanged={onChangeOTP}
              error={error.OTPCodeError && <Text>{I18n.t('OTPError')}</Text>}
              onFocus={() => setError({...error, OTPCodeError: false})}
            />
            <Input
              password
              editable={true}
              label={I18n.t('NewPassword')}
              placeholder={I18n.t('Enterpassword')}
              onChangeText={newInputvalidate}
              error={
                error.newInputError && <Text>{I18n.t('createPassError')}</Text>
              }
              onFocus={() => setError({...error, newInputError: false})}
              createPass={true}
              txt={newInput}
            />
            <Input
              password
              editable={true}
              label={I18n.t('ConfirmNewPassword')}
              placeholder={I18n.t('Enterpassword')}
              onChangeText={confirmInputvalidate}
              error={
                error.confirmInputError && (
                  <Text>{I18n.t('confirmPassError')}</Text>
                )
              }
              onFocus={() => setError({...error, confirmInputError: false})}
              confirmNew={confirmNew}
            />

            <Button
              testID="submit"
              title={I18n.t('Submit')}
              btnWrapper={{...styles.btnCont, ...styles.btnFirst}}
              onPress={redirect}
            />
            <Button
              title={I18n.t('re_send')}
              btnWrapper={{...styles.btnCont, ...styles.btnSecond}}
              btnContainerProp={styles.btnSecondCont}
              btnTxtProp={styles.btnSecondTxt}
              onPress={() => {}}
            />
          </ScrollView>
        </KeyboardAvoidingView>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  mainWrapper: {
    flex: 1,
    backgroundColor: Colors.white,
  },
  subTitleCont: {
    marginVertical: hp(2),
  },
  btnCont: {
    flex: Platform.OS === 'ios' ? 1 : 0,
    justifyContent: Platform.OS === 'ios' ? 'flex-end' : null,
    alignItems: Platform.OS === 'ios' ? 'flex-end' : null,
  },
  btnFirst: {
    marginTop:
      Platform.OS === 'android' && Dimensions.get('window').height < 700
        ? hp(8.5)
        : hp(12.5),
  },
  btnSecond: {
    marginVertical: hp(3),
    marginHorizontal: wp(5),
  },
  btnSecondCont: {
    backgroundColor: Colors.white,
    borderColor: Colors.lightBlue,
    borderWidth: wp(0.3),
  },
  btnSecondTxt: {
    color: Colors.lightBlue,
  },
});
