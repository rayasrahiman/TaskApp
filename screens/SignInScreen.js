import React, {useState, useEffect} from 'react';
import {
  View,
  Text,
  KeyboardAvoidingView,
  Platform,
  Dimensions,
  ScrollView,
  StyleSheet,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import AsyncStorage from '@react-native-async-storage/async-storage';

import Header from '../components/Header';
import Button from '../components/Button';
import Input from '../components/Input';
import TitleAndSubTitle from '../components/TitleAndSubTitle';
import {Colors} from '../constants/colors';
import I18n from '../languages/i18n';

export default function SignInScreen({navigation, route}) {
  const [error, setError] = useState(false);
  const [loader, setLoader] = useState(false);
  const [inputValue, setInputValue] = useState(false);
  const [pass, setPass] = useState('');
  const {email} = route.params;

  const users = async () => {
    const arr = await AsyncStorage.getItem('users');
    console.log(JSON.parse(arr));
  };

  useEffect(() => {
    users();
    if (email) {
      setInputValue(true);
    }
  }, []);

  const onChange = text => {
    setPass(text);
  };

  const redirect = async () => {
    const arr = await AsyncStorage.getItem('users');
    const array = arr ? JSON.parse(arr) : [];
    const user = array.find(item => item.email === email);
    if (user.password === pass) {
      setLoader(true);
      navigation.navigate('Dashboard');
      setLoader(false);
    } else {
      setError(true);
    }
  };

  return (
    <View style={styles.container}>
      <Header />
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={styles.container}>
        <ScrollView>
          <TitleAndSubTitle
            title={I18n.t('signInTitle')}
            subTitle={I18n.t('signInSubTitle')}
            subTitleContProp={styles.subTitleCont}
          />
          <Input
            editable={false}
            label={I18n.t('Email')}
            placeholder={I18n.t('Email')}
            value={email}
            inputValue={inputValue}
            titleHeaderProp={inputValue && {color: Colors.grayishBlack}}
          />
          <Input
            password
            editable={true}
            label={I18n.t('Password')}
            placeholder={I18n.t('Enterpassword')}
            onChangeText={onChange}
            error={error && <Text>{I18n.t('PasswordError')}</Text>}
            onFocus={() => setError(false)}
            passwordLoader={loader}
            forgotPassword={true}
            forgotPass={() => navigation.navigate('ForgotPassword')}
          />

          <Button
            testID="signIn"
            title={I18n.t('Signin')}
            btnWrapper={{...styles.btnCont, ...styles.btnFirst}}
            onPress={redirect}
          />
          <Button
            title={I18n.t('SwitchAccount')}
            btnWrapper={{...styles.btnCont, ...styles.btnSecond}}
            btnContainerProp={styles.btnSecondCont}
            btnTxtProp={styles.btnSecondTxt}
            onPress={() => {}}
          />
        </ScrollView>
      </KeyboardAvoidingView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  subTitleCont: {
    marginVertical: hp(2),
  },
  btnCont: {
    flex: Platform.OS === 'ios' ? 1 : 0,
    justifyContent: Platform.OS === 'ios' ? 'flex-end' : null,
    alignItems: Platform.OS === 'ios' ? 'flex-end' : null,
  },
  btnFirst: {
    marginTop:
      Platform.OS === 'android' && Dimensions.get('window').height < 700
        ? hp(8.5)
        : hp(12.5),
  },
  btnSecond: {
    marginVertical: hp(3),
    marginHorizontal: wp(5),
  },
  btnSecondCont: {
    backgroundColor: Colors.white,
    borderColor: Colors.lightBlue,
    borderWidth: wp(0.3),
  },
  btnSecondTxt: {
    color: Colors.lightBlue,
  },
});
