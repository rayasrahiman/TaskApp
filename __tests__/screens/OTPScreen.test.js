import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';
import {useNavigation} from '@react-navigation/native';

import OTPScreen from '../../screens/OTPScreen';
import SignInScreen from '../../screens/SignInScreen';
import Header from '../../components/Header';
import TitleAndSubTitle from '../../components/TitleAndSubTitle';
import I18n from '../../languages/i18n';

describe('OTPScreen', () => {
  test('should render OTP screen correctly', () => {
    const wrapper = render(<OTPScreen />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render Header', () => {
    const wrapper = shallow(<OTPScreen />);
    wrapper.find(<Header />);
  });

  test('should render TitleAndSubTitle component', () => {
    const wrapper = shallow(<OTPScreen />);
    wrapper.find(<TitleAndSubTitle />);
  });

  test('should render TitleAndSubTitle texts', () => {
    const wrapper = render(<OTPScreen />);
    wrapper.getByText(I18n.t('OTPTitle'));
    wrapper.getByText(I18n.t('OTPSubtitle'));
  });

  test('should render label', () => {
    const wrapper = render(<OTPScreen />);
    wrapper.getByText(I18n.t('Submit'));
    wrapper.getByText(I18n.t('re_send'));
  });

  test('should navigate to SignIn Screen with email after entering valid email address.', async () => {
    const mockedParams = {
      route: {params: {email: 'test@test.com'}},
    };
    render(<OTPScreen />);

    const toClick = await screen.getByTestId('submit');

    fireEvent(toClick, 'press');

    const signInScreen = render(<SignInScreen {...mockedParams} />);

    expect(signInScreen).toBeTruthy();
  });
});
