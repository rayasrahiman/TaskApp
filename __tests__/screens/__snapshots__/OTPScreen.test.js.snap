// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTPScreen should render OTP screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#CCCCCC",
        "borderBottomWidth": 2.5,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assets/Logo.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 112.5,
        }
      }
    />
  </View>
  <RCTScrollView>
    <View>
      <View
        onLayout={[Function]}
      >
        <RCTScrollView>
          <View>
            <View>
              <View
                style={
                  Object {
                    "marginHorizontal": 22.5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#244379",
                      "fontSize": 53.5,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Check your Email
                </Text>
              </View>
              <View
                style={
                  Object {
                    "marginHorizontal": 22.5,
                    "marginVertical": 26.5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#838383",
                      "fontSize": 26.5,
                      "fontWeight": "bold",
                    }
                  }
                >
                  We've sent you an email with a 6-digit code. Please check your email and enter the code below.
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginVertical": 40,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 40,
                    }
                  }
                >
                  6-digit code
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#CCCCCC",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "height": 120,
                        "marginBottom": 66.5,
                        "marginHorizontal": 26.5,
                        "width": 67.5,
                      },
                      false,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="otpInput-0"
                    accessible={true}
                    autoCapitalize="none"
                    clearTextOnFocus={false}
                    firstInput={true}
                    keyboardType="phone-pad"
                    maxLength={6}
                    numberOfInputs={6}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    secureTextEntry={false}
                    selectTextOnFocus={true}
                    style={
                      Object {
                        "fontSize": 40,
                        "textAlign": "center",
                      }
                    }
                    testID="otpInput-0"
                    textContentType="none"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#CCCCCC",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "height": 120,
                        "marginBottom": 66.5,
                        "marginHorizontal": 26.5,
                        "width": 67.5,
                      },
                      false,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="otpInput-1"
                    accessible={true}
                    autoCapitalize="none"
                    autoFocus={false}
                    clearTextOnFocus={false}
                    firstInput={false}
                    keyboardType="phone-pad"
                    maxLength={1}
                    numberOfInputs={6}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    secureTextEntry={false}
                    selectTextOnFocus={true}
                    style={
                      Object {
                        "fontSize": 40,
                        "textAlign": "center",
                      }
                    }
                    testID="otpInput-1"
                    textContentType="none"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#CCCCCC",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "height": 120,
                        "marginBottom": 66.5,
                        "marginHorizontal": 26.5,
                        "width": 67.5,
                      },
                      false,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="otpInput-2"
                    accessible={true}
                    autoCapitalize="none"
                    autoFocus={false}
                    clearTextOnFocus={false}
                    firstInput={false}
                    keyboardType="phone-pad"
                    maxLength={1}
                    numberOfInputs={6}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    secureTextEntry={false}
                    selectTextOnFocus={true}
                    style={
                      Object {
                        "fontSize": 40,
                        "textAlign": "center",
                      }
                    }
                    testID="otpInput-2"
                    textContentType="none"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#CCCCCC",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "height": 120,
                        "marginBottom": 66.5,
                        "marginHorizontal": 26.5,
                        "width": 67.5,
                      },
                      false,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="otpInput-3"
                    accessible={true}
                    autoCapitalize="none"
                    autoFocus={false}
                    clearTextOnFocus={false}
                    firstInput={false}
                    keyboardType="phone-pad"
                    maxLength={1}
                    numberOfInputs={6}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    secureTextEntry={false}
                    selectTextOnFocus={true}
                    style={
                      Object {
                        "fontSize": 40,
                        "textAlign": "center",
                      }
                    }
                    testID="otpInput-3"
                    textContentType="none"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#CCCCCC",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "height": 120,
                        "marginBottom": 66.5,
                        "marginHorizontal": 26.5,
                        "width": 67.5,
                      },
                      false,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="otpInput-4"
                    accessible={true}
                    autoCapitalize="none"
                    autoFocus={false}
                    clearTextOnFocus={false}
                    firstInput={false}
                    keyboardType="phone-pad"
                    maxLength={1}
                    numberOfInputs={6}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    secureTextEntry={false}
                    selectTextOnFocus={true}
                    style={
                      Object {
                        "fontSize": 40,
                        "textAlign": "center",
                      }
                    }
                    testID="otpInput-4"
                    textContentType="none"
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#CCCCCC",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "height": 120,
                        "marginBottom": 66.5,
                        "marginHorizontal": 26.5,
                        "width": 67.5,
                      },
                      false,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="otpInput-5"
                    accessible={true}
                    autoCapitalize="none"
                    autoFocus={false}
                    clearTextOnFocus={false}
                    firstInput={false}
                    keyboardType="phone-pad"
                    maxLength={1}
                    numberOfInputs={6}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    secureTextEntry={false}
                    selectTextOnFocus={true}
                    style={
                      Object {
                        "fontSize": 40,
                        "textAlign": "center",
                      }
                    }
                    testID="otpInput-5"
                    textContentType="none"
                    underlineColorAndroid="transparent"
                  />
                </View>
              </View>
            </View>
            <View>
              <View
                style={
                  Object {
                    "marginHorizontal": 22.5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 26.5,
                      "fontWeight": "bold",
                    }
                  }
                >
                  New Password
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#838383",
                    "borderRadius": 13.5,
                    "borderWidth": 2.5,
                    "flexDirection": "row",
                    "height": 80,
                    "marginHorizontal": 22.5,
                    "marginVertical": 10,
                    "paddingHorizontal": 15,
                  }
                }
              >
                <TextInput
                  autoCorrect={false}
                  editable={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Enter password"
                  secureTextEntry={true}
                  style={
                    Object {
                      "color": "#000000",
                      "flex": 1,
                      "fontSize": 26.5,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    onPress={[Function]}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "fontSize": 40,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰛑
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginHorizontal": 22.5,
                    "marginVertical": 40,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "color": "#CCCCCC",
                          "fontSize": 40,
                          "marginRight": 7.5,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰄬
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#838383",
                        "fontSize": 40,
                      }
                    }
                  >
                    Atleast 8 characters.
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "color": "#CCCCCC",
                          "fontSize": 40,
                          "marginRight": 7.5,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰄬
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#838383",
                        "fontSize": 40,
                      }
                    }
                  >
                    1 uppercase and 1 lowercase letter.
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "color": "#CCCCCC",
                          "fontSize": 40,
                          "marginRight": 7.5,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰄬
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#838383",
                        "fontSize": 40,
                      }
                    }
                  >
                    1 number.
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "color": "#CCCCCC",
                          "fontSize": 40,
                          "marginRight": 7.5,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰄬
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#838383",
                        "fontSize": 40,
                      }
                    }
                  >
                    1 special character.
                  </Text>
                </View>
              </View>
            </View>
            <View>
              <View
                style={
                  Object {
                    "marginHorizontal": 22.5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 26.5,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Confirm New Password
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#838383",
                    "borderRadius": 13.5,
                    "borderWidth": 2.5,
                    "flexDirection": "row",
                    "height": 80,
                    "marginHorizontal": 22.5,
                    "marginVertical": 10,
                    "paddingHorizontal": 15,
                  }
                }
              >
                <TextInput
                  autoCorrect={false}
                  editable={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Enter password"
                  secureTextEntry={true}
                  style={
                    Object {
                      "color": "#000000",
                      "flex": 1,
                      "fontSize": 26.5,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    onPress={[Function]}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Object {
                          "fontSize": 40,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰛑
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "marginHorizontal": 37.5,
                  "marginTop": 167,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0E6E96",
                    "borderRadius": 10,
                    "height": 93.5,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": "100%",
                  }
                }
                testID="submit"
              >
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 40,
                      "paddingVertical": 6.5,
                    }
                  }
                >
                  Submit
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "marginHorizontal": 37.5,
                  "marginVertical": 40,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#0E6E96",
                    "borderRadius": 10,
                    "borderWidth": 2.5,
                    "height": 93.5,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#0E6E96",
                      "fontSize": 40,
                      "paddingVertical": 6.5,
                    }
                  }
                >
                  Re-Send Code
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
